scripts/tasks/sandbox-parts.ts

```diff
@@ -0,0 +1,287 @@
+      throw new Error(dedent`
+          Missing repro directory '${srcDir}'!
+
+          To run sandbox against a local repro, you must have already generated
+          the repro template in the /repros directory using:
+          the repro template in the /repros directory using:
+
+          yarn generate-repros-next --template ${key}
+        `);
+    }
+    await copy(srcDir, sandboxDir);
+  } else {
+    await executeCLIStep(steps.repro, {
+      argument: key,
+      optionValues: { output: sandboxDir, branch: 'next' },
+      cwd: parentDir,
+      dryRun,
+      debug,
+    });
+  }
+
+  const cwd = sandboxDir;
+  for (const addon of addons) {
+    const addonName = `@storybook/addon-${addon}`;
+    await executeCLIStep(steps.add, { argument: addonName, cwd, dryRun, debug });
+  }
+
+  const mainConfig = await readMainConfig({ cwd });
+  mainConfig.setFieldValue(['core', 'disableTelemetry'], true);
+  if (template.expected.builder === '@storybook/builder-vite') forceViteRebuilds(mainConfig);
+  await writeConfig(mainConfig);
+};
+
+export const install: Task['run'] = async ({ sandboxDir }, { link, dryRun, debug }) => {
+  const cwd = sandboxDir;
+
+  await installYarn2({ cwd, dryRun, debug });
+
+  if (link) {
+    await executeCLIStep(steps.link, {
+      argument: sandboxDir,
+      cwd: codeDir,
+      optionValues: { local: true, start: false },
+      dryRun,
+      debug,
+    });
+  } else {
+    // We need to add package resolutions to ensure that we only ever install the latest version
+    // of any storybook packages as verdaccio is not able to both proxy to npm and publish over
+    // the top. In theory this could mask issues where different versions cause problems.
+    await addPackageResolutions({ cwd, dryRun, debug });
+    await configureYarn2ForVerdaccio({ cwd, dryRun, debug });
+
+    await exec(
+      'yarn install',
+      { cwd },
+      {
+        dryRun,
+        startMessage: `⬇️ Installing local dependencies`,
+        errorMessage: `🚨 Installing local dependencies failed`,
+      }
+    );
+  }
+
+  logger.info(`🔢 Adding package scripts:`);
+  await addPackageScripts({
+    cwd,
+    scripts: {
+      storybook:
+        'NODE_OPTIONS="--preserve-symlinks --preserve-symlinks-main" storybook dev -p 6006',
+      'build-storybook':
+        'NODE_OPTIONS="--preserve-symlinks --preserve-symlinks-main" storybook build',
+    },
+  });
+};
+
+// Ensure that sandboxes can refer to story files defined in `code/`.
+// Most WP-based build systems will not compile files outside of the project root or 'src/` or
+// similar. Plus they aren't guaranteed to handle TS files. So we need to patch in esbuild
+// loader for such files. NOTE this isn't necessary for Vite, as far as we know.
+function addEsbuildLoaderToStories(mainConfig: ConfigFile) {
+  // NOTE: the test regexp here will apply whether the path is symlink-preserved or otherwise
+  const loaderPath = require.resolve('../../code/node_modules/esbuild-loader');
```

Comment:
should you use path.join here? not sure if anyone tested this on Windows

Reply:
I'm pretty sure `require.resolve()` takes that into account. I've certainly used it all over the place in other SB code :)

> not sure if anyone tested this on Windows

Would be nice if someone could do that @JReinhold ?